### ISSUE TITLE
Unable to see ChromeCast after reboot

### ISSUE THREAD

**Issue by Nadahar** [jmdns/jmdns#231]
I've been trying to figure out what goes wrong for a little while, but exactly what goes wrong eludes me.

I'm working on a media server that uses jmdns to discover "Google Cast" devices like the ChromeCast. It mostly works fine, but I've discovered that if the ChromeCast device reboots, jmdns sends the `remove` event, but doesn't send a new `add` or `registered` event when the device comes back on line. I've done many attempts at studying this, and it seems that it will eventually be "rediscovered", but I can't find any logic in when that is. I have experienced anything from 4 minutes to 3 hours and 20 minutes before it is "rediscovered".

I've tried to dive into the details, I've read through the entire RFC6762 and "selected parts" of RFC6763, but to no avail. I have problems interpreting jmdns' logging enough to understand what's really going on, but it *looks* to me like the ChromeCast announces itself like it should when it's coming back up, and it also seems like it's re-added to jmdns' cache, but no `add` or `registered` events are sent. This indicates to me that the problem is most likely with jmdns, but I don't understand the log well enough to be sure.

I'm attaching a log file where I've manually "isolated" one such reboot by deleting log events that doesn't come from jmdns or isn't directly related. I've made a "debug implementation" of both listener types that will merely log the event that is received, and I've left these in the log. The entries from the "debug listeners" can be recognized by `net.pms.service.MulticastDNS`.

I think this problem should be easily reproducible by anybody that has a "cast device", which includes ChromeCasts, Google Home devices and Android TVs. All I do to trigger the reboot is to use the "Google Home" application on Android and send the command to reboot from there.

I hope somebody that know this better than me can help analyze the situation, as I seem to be stuck.

[cast_reboot_log.txt](https://github.com/jmdns/jmdns/files/6957652/cast_reboot_log.txt)

**Comment by melloware** [comment]
Does this still happen with 3.6.1?

**Comment by Nadahar** [comment]
I have no idea, it has been 4 years since I worked on this. It never "stopped" being like that as long as I worked on it.

**Comment by melloware** [comment]
Ok can you please close the ticket then.

**Comment by Nadahar** [comment]
Has it been fixed? Others can test this just as easy as myself.

**Comment by melloware** [comment]
I just asked you to test the latest version and report back if it's still an issue. You said you haven't used it in 4 years so I took that as you no longer care so let's not leave this ticket open indefinitely.

**Comment by Nadahar** [comment]
No, it's not that "I don't care", it's just that I don't remember the things I did then, and I don't have this project checked out/set up for development at the moment. So, I could test it, but it would require potentially a lot of work on my part to get "up to speed" again.

If nothing has been done to address the issue, why should I do all this just to find out that nothing has changed? I don't remember the little I knew about mDNS back then, but I think this is a pretty "fundamental" thing where the standard describes how it should be handled (that a device goes away and comes back). What I do remember is that when I tested other "mDNS clients" they rediscover the cast device as soon as it boots up again, so the issue seems to be with jmdns.

**Comment by Nadahar** [comment]
I'll have to try to find the motivation to dive back into it again then. I'm not sure that you need a Chromecast for this, it's just where I experienced it. It might apply to many devices that are removed and then re-added shortly thereafter. Without knowing what the exact issue is, it's impossible to know, but I wouldn't assume that this is/was a "Chromecast only" issue.

**Comment by melloware** [comment]
Maybe you are correct but I have been using this lib since 2009 without issue so clearly none of the devices I have produce this issue. 

You can reproduce it with a chromecast so let's focus on something we know reproduces the issue right?

Also this board is not flooded with issues around this particular issue so I wonder if it's something chromecast specific.
